src/coreclr/jit/rationalize.cpp

```diff
@@ -574,8 +574,12 @@ Compiler::fgWalkResult Rationalizer::RewriteNode(GenTree** useEdge, Compiler::Ge
             // args as these have now been sequenced.
             for (CallArg& arg : node->AsCall()->gtArgs.EarlyArgs())
             {
-                if (!arg.GetEarlyNode()->IsValue())
+                if (arg.GetLateNode() != nullptr)
                 {
+                    if (arg.GetEarlyNode()->IsValue())
```

Comment:
Can you just make the comma you create in your pass `TYP_VOID`?

Reply:
Well, not without checking `gtGetParent` I guess.

Reply:
I don't think it is possible in the general case for the check here to remain the same. After we delete the store, the only side-effecting RHS tree could be a value (e. g. `ASG(LCL_FLD, CALL)`).